### PR TITLE
Get ordering from model in module controller

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
The module ordering controller task which is used in the ordering form field should use the model to fetch the modules for the same client and position.

### Testing Instructions
Edit a module and click on the order form field.

### Actual result BEFORE applying this Pull Request
The modules for the same client and position are listed.

### Expected result AFTER applying this Pull Request
The modules for the same client and position are listed.

### CODE REVIEW

File: administrator/components/com_modules/src/Controller/ModuleController.php [hunk 1]
```diff
@@ -289,22 +289,14 @@ public function orderPosition()
 			$app->close();
 		}
 
-		$db    = Factory::getDbo();
```
Comment (Quy) [inline administrator/components/com_modules/src/Controller/ModuleController.php]: Please remove imports.

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 9d8af1d1c0f8b1433ec427b5a41c07b32b1d683a<br /><br />Tested successfully in Joomla 4.2-alpha3-dev of 29 April in Wampserver 3.2.8 using PHP 8.0.15.

Had to change Error Reporting in Global Configuration from 'Maximum' to 'Simple' to get the Ordering info displayed, because of the PHP notice:
Notice: Only variable references should be returned by reference in C:\www\joomla42\libraries\src\MVC\Model\BaseDatabaseModel.php on line 418<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37635">issues.joomla.org/tracker/joomla-cms/37635</a>.</sub>

**Comment by laoneo** [comment]
Can you check if your branch is up to date with 4.2?

**Comment by roland-d** [comment]
@laoneo This looks like something is broken in the PR, the list looks like this for me
image

Notice that the list restarts at 1 and this happens 3 times but the list is some other data every time.

**Comment by laoneo** [comment]
@roland-d can you test this again as I can't reproduce the issue at all?

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 099307624d7d5aec6123647407ada4cc2881c25a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37635">issues.joomla.org/tracker/joomla-cms/37635</a>.</sub>

**Pull request merged**